rwengine/src/loaders/LoaderGXT.cpp

```diff
@@ -8,20 +8,21 @@
 #include <platform/FileHandle.hpp>
 
 #include "data/GameTexts.hpp"
+#include "rw/bit_cast.cpp"
 
 void LoaderGXT::load(GameTexts &texts, FileHandle &file) {
     auto data = file->data;
 
     data += 4;  // TKEY
 
-    std::uint32_t blocksize = *(std::uint32_t *)data;
+    std::uint32_t blocksize = bit_cast<std::uint32_t>(*data);
 
     data += 4;
 
     auto tdata = data + blocksize + 8;
 
     for (size_t t = 0; t < blocksize / 12; ++t) {
-        size_t offset = *(std::uint32_t *)(data + (t * 12 + 0));
+        size_t offset = bit_cast<std::uint32_t>(data + (t * 12 + 0));
```

Comment:
nit: who thought it was a good idea to do (presumably) pointer arithmetic on an `auto` variable?

Reply:
This whole thing seems a bit smelly IMO.. would it not be better to have a BinaryStream object or something that you can Seek/Read(T) etc instead?

Reply:
Do you think about treating it as stream?